Mathlib/CategoryTheory/ConcreteCategory/Basic.lean

```diff
@@ -204,15 +227,196 @@ def HasForget₂.trans (C : Type u) [Category.{v} C] [ConcreteCategory.{w} C]
 
 /-- Every forgetful functor factors through the identity functor. This is not a global instance as
     it is prone to creating type class resolution loops. -/
-def hasForgetToType (C : Type u) [Category.{v} C] [ConcreteCategory.{w} C] :
+def hasForgetToType (C : Type u) [Category.{v} C] [HasForget.{w} C] :
     HasForget₂ C (Type w) where
   forget₂ := forget C
   forget_comp := Functor.comp_id _
 
 @[simp]
-lemma NatTrans.naturality_apply {C D : Type*} [Category C] [Category D] [ConcreteCategory D]
+lemma NatTrans.naturality_apply {C D : Type*} [Category C] [Category D] [HasForget D]
     {F G : C ⥤ D} (φ : F ⟶ G) {X Y : C} (f : X ⟶ Y) (x : F.obj X) :
     φ.app Y (F.map f x) = G.map f (φ.app X x) := by
   simpa only [Functor.map_comp] using congr_fun ((forget D).congr_map (φ.naturality f)) x
 
+section ConcreteCategory
+
+/-- A concrete category is a category `C` where objects correspond to types and morphisms to
+(bundled) functions between those types.
+
+In other words, it has a fixed faithful functor `forget : C ⥤ Type`.
+
+Note that `ConcreteCategory` potentially depends on three independent universe levels,
+* the universe level `w` appearing in `forget : C ⥤ Type w`
+* the universe level `v` of the morphisms (i.e. we have a `Category.{v} C`)
+* the universe level `u` of the objects (i.e `C : Type u`)
+They are specified that order, to avoid unnecessary universe annotations.
+-/
+class ConcreteCategory (C : Type u) [Category.{v} C]
+    (FC : outParam <| C → C → Type*) {CC : outParam <| C → Type w}
```

Comment:
I noticed that `CC` can always be inferred through unification in the `FunLike` instance. So for conciseness I made it implicit. I was not expecting that to be possible!